pkg/smokescreen/smokescreen.go

```diff
@@ -655,14 +655,19 @@ func remapHost(config *Config, destination *hostport.HostPort, sctx *Smokescreen
 	if config.HostRemapConfig == nil {
 		return
 	}
-	for originalHost, newHost := range config.HostRemapConfig.Mappings {
+	// TODO(xwen-figma): reduce log verbosity after testing.
+	for originalHost, newHostEntry := range config.HostRemapConfig.Mappings {
+		if newHostEntry.Enabled == nil {
+			sctx.logger.Warnf("Enabled function is nil for: %s, remapping is disabled", originalHost)
+			continue
+		}
+		if newHostEntry.NewHost == "" || !newHostEntry.Enabled() {
```

Comment:
I feel the `newHostEntry.Enabled()` should only be called within the `strings.Contains` below, otherwise we end up calling Enabled() for every item in the map.